### PR TITLE
use CompletionStage instead of Future in java doc classes

### PR DESCRIPTION
**Pull request by ortibazar**
Refs #22393

Removed unused scala future import lines in some java files and converted Future to CompletionStage

### CODE REVIEW

**Comment by ortigali** [comment]
ActorDocTest.java was not refactored. After conversion code looks ugly IMO (especially ask pattern part)

File: akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ private void receiveMsg3(Msg3 msg) {
     public void onReceive(Object msg) throws Exception {
       if (msg instanceof Msg1)
         receiveMsg1((Msg1) msg);
-      else if (msg instanceof Msg1)
+      else if (msg instanceof Msg2)
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java]: good catch

Comment (ktoso) [inline akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java]: Actually, why do we have code that still uses UntypedAbstractActor? cc @patriknw

Comment (ortigali) [inline akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java]: http://doc.akka.io/docs/akka/2.5-M2/project/migration-guide-2.4.x-2.5.x.html#UntypedActor

**Review by ktoso — APPROVED** [review]
LGTM

File: akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java]: Ah great, had such suspicion :) Makes sense

**Comment by ktoso** [comment]
> ActorDocTest.java was not refactored. After conversion code looks ugly IMO (especially ask pattern part)

Well, it is Java... Could you clarify why it gets ugly? Should not be much uglier than the usual

**Comment by ortigali** [comment]
PLS BUILD

File: akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java (hunk 1 above)
Comment (patriknw) [inline akka-docs/rst/java/code/docs/actorlambda/ActorDocTest.java]: yes, this is not `UntypedActor`, it just looks like such.
`OptimizedActor extends UntypedAbstractActor`

File: akka-docs/rst/java/code/docs/pattern/SupervisedAsk.java [hunk 2]
```diff
@@ -61,13 +58,10 @@ public Receive createReceive() {
...
-          new Function<Throwable, Directive>() {
-            public Directive apply(Throwable cause) {
-              caller.tell(new Status.Failure(cause), self());
-              return SupervisorStrategy.stop();
-            }
-          });
+      return new OneForOneStrategy(0, Duration.Zero(), cause -> {
+          caller.tell(new Status.Failure(cause), self());
+          return SupervisorStrategy.stop();
+        });
```
Comment (2m) [inline akka-docs/rst/java/code/docs/pattern/SupervisedAsk.java]: Of course not part of this PR, but this reminds me that we should get #20891 going, to have consistent formatting in java files.

**Review by 2m — APPROVED** [review]

**Comment by 2m** [comment]
Was the `ActorDocTest.java` ask part verbose after refactor because it uses `Futures.sequence`? If that was the case you could use `CompletableFuture.allOf` to achieve the same.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
looks like ActorDocTest was also fixed, so I'm merging

**Comment by patriknw** [comment]
great work as usual, @ortigali

**Comment by patriknw** [comment]
@ortigali BTW are there corresponding leftover in https://github.com/akka/akka-samples ?

**Pull request merged**